### ISSUE TITLE
padding-line-between-statements: add multiline-return

### ISSUE THREAD

**Issue by scola84** [eslint/eslint#14772]
**What rule do you want to change?**
`padding-line-between-statements`

**Does this change cause the rule to produce more or fewer warnings?**
More (depending on the state of the code).

**How will the change be implemented? (New option, new default behavior, etc.)?**
A new `STATEMENT_TYPE` should be added, e.g. `multiline-return`, it would probably work like the other `multiline-*` types.

**Please provide some example code that this change will affect:**

```js
function doSomething() {
  let xyz = '123'
  return xyz === '123'
    ? null
    : xyz
}

// becomes

function doSomething() {
  let xyz = '123'

  return xyz === '123'
    ? null
    : xyz
}
```

**What does the rule currently do for this code?**
It always removes blank lines before return statements, whether they are singleline or multiline.

**What will the rule do after it's changed?**
It will keep a blank line in the case of a multiline return statement.

**Are you willing to submit a pull request to implement this change?**
Not at the moment.

**Comment by aladdin-add** [comment]
Per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), we can only accept add/change rule if they relate to new ECMAScript features that reached stage 4 in the preceding 12 months, so this proposal doesn't qualify. You can always make a custom rule and publish a plugin.
